src/clp_ffi_js/ir/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,233 @@
+                break;
+            }
+            if (std::errc::result_out_of_range == error) {
+                SPDLOG_ERROR("File contains an incomplete IR stream");
+                break;
+            }
+            throw ClpJsException(
+                    clp::ErrorCode::ErrorCode_Corrupt,
+                    __FILENAME__,
+                    __LINE__,
+                    "Failed to decompress: "s + error.category().name() + ":" + error.message()
+            );
+        }
+        m_data_buffer.reset(nullptr);
+    }
+
+    auto results{emscripten::val::object()};
+    results.set("numValidEvents", m_log_events.size());
+    results.set("numInvalidEvents", 0);
+    return results;
+}
+
+auto ClpIrV1Decoder::decode(size_t begin_idx, size_t end_idx) const -> emscripten::val {
+    if (m_log_events.size() < end_idx || begin_idx >= end_idx) {
+        return emscripten::val::null();
+    }
+
+    size_t log_num{begin_idx + 1};
+    std::string message;
+    constexpr size_t cDefaultReservedMessageLength{512};
+    message.reserve(cDefaultReservedMessageLength);
+    auto const results{emscripten::val::array()};
+    std::span<clp::ir::LogEvent<clp::ir::four_byte_encoded_variable_t> const> const log_events_span(
+            m_log_events.begin() + static_cast<std::ptrdiff_t>(begin_idx),
+            m_log_events.begin() + static_cast<std::ptrdiff_t>(end_idx)
+    );
+    for (auto const& log_event : log_events_span) {
+        message.clear();
+
+        // TODO: Replace below handlers code by an OSS decoding method once it's added in the future
+        auto const constant_handler = [&](std::string const& value, size_t begin_pos, size_t length
+                                      ) { message.append(value, begin_pos, length); };
+
+        auto const encoded_int_handler = [&](clp::ir::four_byte_encoded_variable_t value) {
+            message.append(clp::ffi::decode_integer_var(value));
+        };
+
+        auto const encoded_float_handler
+                = [&](clp::ir::four_byte_encoded_variable_t encoded_float) {
+                      message.append(clp::ffi::decode_float_var(encoded_float));
+                  };
+
+        auto const dict_var_handler
+                = [&](std::string const& dict_var) { message.append(dict_var); };
+
+        try {
+            clp::ffi::ir_stream::generic_decode_message<true>(
+                    log_event.get_logtype(),
+                    log_event.get_encoded_vars(),
+                    log_event.get_dict_vars(),
+                    constant_handler,
+                    encoded_int_handler,
+                    encoded_float_handler,
+                    dict_var_handler
+            );
+        } catch (clp::ffi::ir_stream::DecodingException const& e) {
+            SPDLOG_ERROR("Failed to decode. Error Code: {}", e.get_error_code());
+            break;
+        }
+
+        constexpr size_t cLogLevelNone{0};
+        constexpr size_t cLogLevelBeginIdx{cLogLevelNone + 1};
```

Comment:
This should probably be `cValidLogLevelsBeginIdx` and be with the log level constants.